pkg/volume/plugins.go

```diff
@@ -235,6 +235,7 @@ type Spec struct {
 	Volume           *api.Volume
 	PersistentVolume *api.PersistentVolume
 	ReadOnly         bool
+	Namespace        string
```

Comment:
Instead of adding the `Namespace` field in the `Spec struct` let's just pass it directly to `Attach()` or `NewAttacher()`. Because `Spec` is used in a lot of places, and most `Spec` consumers don't care about the `Namespace`, but worse we don't have the `Namespace` available in the places the `Spec` is constructed so we end up sticking empty string in there. I'm worried that some consumer somewhere may use the field when it is not correctly set.